cdc/processor.go

```diff
@@ -654,35 +660,66 @@ func (p *processor) sinkDriver(ctx context.Context) error {
 
 // syncResolved handle `p.ddlJobsCh` and `p.resolvedTxns`
 func (p *processor) syncResolved(ctx context.Context) error {
-	defer log.Info("syncResolved stopped")
-	defer p.sinkEmittedResolvedReceiver.Stop()
-	metricWaitPrepare := waitEventPrepareDuration.WithLabelValues(p.changefeedID, p.captureID)
+	defer func() {
+		p.sinkEmittedResolvedReceiver.Stop()
+		log.Info("syncResolved stopped")
+	}()
+
+	events := make([]*model.PolymorphicEvent, 0, defaultSyncResolvedBatch)
+
```

Comment:
```suggestion
emitEvs := make([]*model.RowChangedEvent, 0, defaultSyncResolvedBatch)
```
and remove L670
and add `emitEvs = emitEvs[:0]` to L686

Reply:
addressed